src/NLog/NestedDiagnosticsContext.cs

```diff
@@ -162,8 +162,8 @@ public static object[] GetAllObjects()
         /// </summary>
         private class StackPopper : IDisposable
         {
-            private Stack<object> _stack;
-            private int _previousCount;
+            private readonly Stack<object> _stack;
+            private readonly int _previousCount;
```

Comment:
note that `readonly` could influence runtime performance (badly)

Reply:
Only heard that it affect performance for fields in struct, where it will cause the entire struct to be copied on field access. Do you have some details about this other readonly pitfall?

Reply:
this one, but that's indeed structs 

https://codeblog.jonskeet.uk/2014/07/16/micro-optimization-the-surprising-inefficiency-of-readonly-fields/